app/src/main/java/com/example/myapplication/news/NewsScreen.kt

```diff
@@ -65,26 +77,28 @@ class NewsScreen : Fragment() {
         }
 
         if (savedInstanceState != null) {
-            list = savedInstanceState.getParcelableArrayList<NewsItem>(NEWS_LIST) as ArrayList<NewsItem>
+            list = savedInstanceState.getParcelableArrayList<NewsItem>(NEWS_LIST) as List<NewsItem>
             binding.newsProgress.visibility = View.GONE
             adapter.differ.submitList(filterNews())
         } else {
-            Observable
-                .fromCallable{JSONReader(requireContext(), resources.getString(R.string.news_file), NewsItem::class.java).getList()}
-                .subscribeOn(Schedulers.newThread())
+            Common.retrofitServices.getNewsList()
```

Comment:
Сейчас у тебя неправильно обрабатывается success состояние, просто сетится лист, но данные не будут отображаться. Замени всю цепочку на это
```suggestion
             Common.retrofitServices.getNewsList()
                .subscribeOn(Schedulers.computation())
                .observeOn(AndroidSchedulers.mainThread())
                .onErrorResumeNext { Single.fromCallable {
                    JSONReader(requireContext(), resources.getString(R.string.news_file), NewsItem::class.java).getList().toList()
                } }
                .subscribe(
                    { items ->
                        list = items
                        binding.newsProgress.visibility = View.GONE
                        adapter.differ.submitList(filterNews())
                    },
                    {
                        list = JSONReader(requireContext(), resources.getString(R.string.news_file), NewsItem::class.java).getList()
                        binding.newsProgress.visibility = View.GONE
                        adapter.differ.submitList(filterNews())
                    }
                )
```